src/components/scene/auto-enter-vr.js

```diff
@@ -0,0 +1,53 @@
+var registerComponent = require('../../core/component').registerComponent;
+var utils = require('../../utils');
+
+/**
+ * Automatically enter VR, either upon vrdisplayactivate (e.g. putting on Rift headset)
+ * or immediately (if possible) if display name contains data string.
+ * The default data string is 'GearVR' for Carmel browser which only does VR.
+ */
+module.exports.Component = registerComponent('auto-enter-vr', {
+  schema: {default: 'GearVR'},
+
+  init: function () {
+    var scene = this.el;
+
+    // define methods to allow mock testing
+    this.enterVR = scene.enterVR.bind(scene);
+    this.exitVR = scene.exitVR.bind(scene);
+    this.shouldAutoEnterVR = this.shouldAutoEnterVR.bind(this);
+
+    // don't do anything if false
+    if (utils.getUrlParameter('auto-enter-vr') === 'false') { return; }
+
+    // enter VR on vrdisplayactivate (e.g. putting on Rift headset)
+    window.addEventListener('vrdisplayactivate', function () { this.enterVR(); }, false);
+
+    // exit VR on vrdisplaydeactivate (e.g. taking off Rift headset)
+    window.addEventListener('vrdisplaydeactivate', function () { this.exitVR(); }, false);
+
+    // check if we should try to enter VR... turns out we need to wait for next tick
+    var self = this;
+    setTimeout(function () { if (self.shouldAutoEnterVR()) { this.enterVR(); } }, 0);
```

Comment:
not exactly sure, but it appears that the renderer etc. needs to be set up for this to succeed, which isn't ready to go at init